pkg/preparer/orchestrate.go

```diff
@@ -79,6 +79,89 @@ func (p podWorkerID) String() string {
 	return fmt.Sprintf("%s-%s", p.podID.String(), p.podUniqueKey)
 }
 
+func (p *Preparer) InstallWhiteListPods(whiteListPods map[types.PodID]bool, preparerConfig *PreparerConfig) error {
+	intentResults, _, err := p.store.ListPods(consul.INTENT_TREE, p.node)
+	realityResults, _, err := p.store.ListPods(consul.REALITY_TREE, p.node)
+	var intents []consul.ManifestResult
+	if err != nil {
+		p.Logger.WithError(err).Errorln("could not check intent")
```

Comment:
1) The error here may be from checking reality, so the error message is misleading.
2) Should we return the error here? An inability to check intent seems fatal worthy given the intention of this function. As it stands, I believe this function would return nil which might mislead the caller into thinking this function installed the white listed pods (as one would naturally expect given the name and the nil return value)

Reply:
fixed

Reply:
`err` is set to a new value on line 84 and its value on line 83 is never read or consumed. If there is an error reading intent, but not reality, `err` will be nil even though there _was_ an error.

Reply:
good catch!